Volume/Supporting/Main.swift

```diff
@@ -32,6 +34,19 @@ struct Main: App {
             announcementPath: Secrets.announcementsPath
         )
     }
+    
+    private func registerForPushNotifications() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    print("Push notification permissions granted: \(granted)")
+                }
+            }
+        }
+        DispatchQueue.main.async {
+            UIApplication.shared.registerForRemoteNotifications()
```

Comment:
This should be inside of `getNotificationSettings`, both if notifs are authorized and after requesting authorization succeeds.